File: Data Structures & Algorithms/longest-consecutive-sequence/submission-4.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Map<Integer, Integer> map = new HashMap<>();
        int length = 0;

        for (int num : nums)
        {
            if (!map.containsKey(num))
            {
                map.put(num, map.getOrDefault(num - 1, 0) + map.getOrDefault(num + 1, 0) + 1);
                map.put(num - map.getOrDefault(num - 1, 0), map.get(num));      // left boundary = length
                map.put(num + map.getOrDefault(num + 1, 0), map.get(num));      // right boundary = length

                length = Math.max(length, map.get(num));        // get the largest number
            }
        }

        return length;
    }
}

File: Data Structures & Algorithms/concatenation-of-array/submission-4.py
class Solution:
    def getConcatenation(self, nums: List[int]) -> List[int]:
        new_nums = []

        for i in range(0, len(nums)):
            for j in range(0, 1):
                new_nums.append(nums[i])

        return new_nums*2
        
